### PR TITLE
splitting the zip test suite

### LINKED ISSUE #34927: Split the quarkus zip test suite
To get the github action runtime of the quarkus dist zip tests under 60 minutes we will need to split the tests into 2 runs.

### PR DESCRIPTION
**Pull request by shawkins**
splits the zip tests into those that run quickly and slowly. Also uses tags to control what is run.

closes: #34927

### CODE REVIEW

**Comment by shawkins** [comment]
@mabartos @vmuzikar @Pepo48 this should split our zip test suites into roughly equal parts. I went with a fast and slow convention - but we can use whatever terms make sense. We just need to make sure they remain balanced over time. Switching to tags helps make what falls into which test suite a little easier to maintain.

**Review by mabartos — COMMENTED** [review]
@shawkins Nice job! I like the idea of using tags. I just add a small comment below for better DevEx :) Other than that, LGTM. This is a very huge improvement in the CI exec time :tada: :tada: 

slow=~27m
fast=~26m

And what will be the process around determining whether the test should be included in `slow`, or `fast`? Basically, if it uses a lot of augmentation, we should mark it as `slow`? I'm just curious how we will proceed in future as one of these categories will grow. But I'm okay to have it like that and occasionally reanalyze new tests or balance them.

**Comment by shawkins** [comment]
> And what will be the process around determining whether the test should be included in `slow`, or `fast`? Basically, if it uses a lot of augmentation, we should mark it as `slow`? I'm just curious how we will proceed in future as one of these categories will grow. But I'm okay to have it like that and occasionally reanalyze new tests or balance them.

The distinction is mostly arbitrary as the goal is simply to divide the execution time. My thinking was that the "fast" tests give you the most test classes in a given amount of time, but that may not be of any practical use. Once one of these categories grow, someone will need to rebalance - add or remove the slow tag depending on which test suite is taking too long.

**Review by mabartos — APPROVED** [review]
@shawkins LGTM, thanks!

PS: It'd be good to introduce some profiles for them as you mentioned - in a follow up task.

**Comment by shawkins** [comment]
> PS: It'd be good to introduce some profiles for them as you mentioned - in a follow up task.

I wasn't completely sold on doing that just yet - all we'd be doing is setting a single property, so it's not encapsulating much.

**Comment by Pepo48** [comment]
@shawkins LGTM, I like this idea. I can imagine we could spend time with endless discussions about what belongs where and there would be no guarantee of a more practical distribution anyway. I consider keeping one half should as diverse as possible, while keeping both halves roughly the same timewise to be actually very practical approach, especially as a starting point.

Thanks!

**Review by Pepo48 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -289,14 +291,17 @@ jobs:
...
       - name: Run Quarkus integration Tests
         run: |
           declare -A PARAMS
-          PARAMS["win"]="-Dtest=JavaOptsScriptTest,StartCommandDistTest,StartDevCommandDistTest,BuildAndStartDistTest,ImportAtStartupDistTest"
-          PARAMS["zip"]=""
+          PARAMS["win"]="-Dkc.quarkus.tests.groups=win"
+          PARAMS["zip-slow"]="-Dkc.quarkus.tests.groups=slow"
+          PARAMS["zip-fast"]='-Dkc.quarkus.tests.groups=!slow'
           PARAMS["container"]="-Dkc.quarkus.tests.dist=docker"
-          PARAMS["storage"]="-Ptest-database -Dtest=PostgreSQLDistTest,MariaDBDistTest#testSuccessful,MySQLDistTest#testSuccessful,DatabaseOptionsDistTest,JPAStoreDistTest,HotRodStoreDistTest,MixedStoreDistTest,TransactionConfigurationDistTest,ExternalInfinispanTest"
```
Comment (vmuzikar) [inline .github/workflows/ci.yml:298]: Are we missing the storage tag on the following tests?
JPAStoreDistTest,HotRodStoreDistTest,MixedStoreDistTest,ExternalInfinispanTest

**Review by vmuzikar — COMMENTED** [review]
I like the idea of using tags, thanks for that @shawkins. Overall LGTM, just one minor comment.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (shawkins) [reply .github/workflows/ci.yml:298]: No, those tests no longer exist.

**Review by vmuzikar — APPROVED** [review]
LGTM!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (shawkins) [reply .github/workflows/ci.yml:298]: No problem. This approach should mean that we won't have this problem moving forward.

**Pull request merged**